### PR TITLE
chore: upgrade to google.golang.org/grpc v1.68.0

### PR DESCRIPTION
**Pull request by vandjelk**
**Description**

Changes proposed in this pull request:

- Upgrade google.golang.org/grpc v1.68.0

Steps:
- modified go.mod to use google.golang.org/grpc v1.68.0
- run go mod tidy

go mod tidy added toolchain directive which was causing failures in go lint check.
after removing toolchain directive from go.mod all checks passed.

### CODE REVIEW

File: .github/workflows/pr-golang-lint.yml [hunk 1]
```diff
@@ -15,9 +15,11 @@ jobs:
...
         with:
-          go-version: '1.21'
+          go-version: '1.22.7'
       - name: golangci-lint
         uses: golangci/golangci-lint-action@v6.0.1
         with:
           version: v1.59
           args: --timeout=10m
+          only-new-issues: true
+          skip-cache: true
```
Comment (dushanpantic) [inline .github/workflows/pr-golang-lint.yml:25]: why was this added?

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module github.com/kyma-project/cloud-manager
 
-go 1.22.5
+go 1.22.7
```
Comment (dushanpantic) [inline go.mod:3]: Why is Go upgraded?

Comment (vandjelk) [reply go.mod:3]: change go.mod `google.golang.org/grpc v1.65.0 -> google.golang.org/grpc v1.68.0`
run `go mod tidy`

**Review by dushanpantic — APPROVED** [review]
